web/packages/teleport/src/Main/LayoutContext.tsx

```diff
@@ -21,20 +21,37 @@ import React, {
   PropsWithChildren,
   useContext,
   useState,
+  useEffect,
 } from 'react';
 
 interface LayoutContextValue {
   hasDockedElement: boolean;
   setHasDockedElement: (value: boolean) => void;
+  currentWidth: number;
 }
 
 const LayoutContext = createContext<LayoutContextValue>(null);
 
 export function LayoutContextProvider(props: PropsWithChildren<unknown>) {
   const [hasDockedElement, setHasDockedElement] = useState(false);
+  const [currentWidth, setCurrentWidth] = useState(window.innerWidth);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setCurrentWidth(window.innerWidth);
+    };
+
+    window.addEventListener('resize', handleResize);
```

Comment:
Instead of an event listener on the `resize` event, I would recommend using [the more performant](https://tigeroakes.com/posts/resize-event-alternatives/) `ResizeObserver`. You can look at an example of how I used it [here](https://github.com/gravitational/teleport/blob/181d69b21f40658b5d0ccd143bd649b91b8fa99b/web/packages/shared/components/UnifiedResources/UnifiedResources.tsx#L372-L386).